app/models/concerns/user_authentication_concern.rb

```diff
@@ -29,6 +32,10 @@ def email_required?
     true
   end
 
+  def create_instance_users
```

Comment:
Where are the reset of instance methods ?  I already put them under other instance methods in this concern.